### PR TITLE
Add PI2 I2C support

### PR DESCRIPTION
**Pull request by JustAddWires**
Re-factor I2C classes to align with support for a second controller type
Rename one of two different uses of the name "m_controller" to avoid confusion
Fix some typos

### CODE REVIEW

File: source/BtI2cController.h [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+    virtual ~BtI2cControllerClass()
+    {
+        this->end();
+    }
+    
+    // Initialize the specified I2C bus for use.
+    HRESULT begin(ULONG busNumber) override;
+
+    // This method returns the external I2C bus pins to their default configurations.
+    void end() override
```
Comment (zfields) [inline source/BtI2cController.h]: inline or move with begin() to prevent possible problem

File: source/CY8C9540ASupport.cpp [hunk 2]
```diff
@@ -137,7 +138,7 @@ HRESULT CY8C9540ADevice::SetBitState(ULONG i2cAdr, ULONG portBit, ULONG state)
     if (SUCCEEDED(hr))
     {
         // Actually perform the I2C transfers specified above.
-        hr = transaction.execute();
+        hr = transaction.execute(g_i2c.getController());
```
Comment (zfields) [inline source/CY8C9540ASupport.cpp]: consistency with ADS1015Support

File: source/I2c.h [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    
+    /// Prepare to use the I2C controller associated with this object.
+    HRESULT begin();
+
+    /// Finish using the I2C controller associated with this object.
+    void end();
+
+    /// Method to get the address of the I2C Controller object.
+    I2cControllerClass* getController()
```
Comment (zfields) [inline source/I2c.h]: inline?

File: source/I2cTransaction.h [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        m_pXfrQueueTail(nullptr),
+        m_hI2cLock(INVALID_HANDLE_VALUE),
+        m_abort(FALSE),
+        m_error(SUCCESS),
+        m_isIncomplete(FALSE),
+        m_useHighSpeed(FALSE)
+    {
+    }
+
+    virtual ~I2cTransactionClass()
```
Comment (zfields) [inline source/I2cTransaction.h]: inline?

File: source/I2cTransaction.h [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+    // Prepare this transaction for re-use.
+    // Any previously set slave address is not affected by this method.
+    void reset();
+
+    // Sets the 7-bit address of the slave for this transaction.
+    HRESULT setAddress(ULONG slaveAdr);
+
+    // Gets the 7-bit address of the slave for this transaction.
+    ULONG getAddress()
```
Comment (zfields) [inline source/I2cTransaction.h]: inline?

File: source/I2cTransfer.h [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+        else
+        {
+            PUCHAR nextRead = &(m_pBuffer[m_nextRead]);
+            m_nextRead++;
+            return nextRead;
+        }
+    }
+
+    // Method to associate a callback routine with this transfer.
+    HRESULT setCallback(std::function<HRESULT()> callBack)
```
Comment (zfields) [inline source/I2cTransfer.h]: inline?

File: source/PCA9685Support.cpp [hunk 7]
```diff
@@ -92,7 +94,7 @@ HRESULT PCA9685Device::SetBitState(ULONG i2cAdr, ULONG portBit, ULONG state)
     if (SUCCEEDED(hr))
     {
         // Actually perform the I2C transfers specified above.
-        hr = transaction.execute();
+        hr = transaction.execute(g_i2c.getController());
```
Comment (zfields) [inline source/PCA9685Support.cpp]: consistency with ADS1015Support

File: source/PCAL9535ASupport.cpp [hunk 8]
```diff
@@ -4,7 +4,9 @@
 
 #include "PCAL9535ASupport.h"
 #include "ExpanderDefs.h"
-#include "I2cController.h"
+#include "I2c.h"
```
Comment (zfields) [inline source/PCAL9535ASupport.cpp]: consistency with ADS1015Support?

File: source/I2c.h (hunk 3 above)
Comment (JustAddWires) [inline source/I2c.h]: Fixed.

File: source/I2cTransaction.h (hunk 4 above)
Comment (JustAddWires) [inline source/I2cTransaction.h]: Fixed.

File: source/I2cTransaction.h (hunk 5 above)
Comment (JustAddWires) [inline source/I2cTransaction.h]: Fixed.

File: source/I2cTransfer.h (hunk 6 above)
Comment (JustAddWires) [inline source/I2cTransfer.h]: Fixed.

File: source/BtI2cController.h (hunk 1 above)
Comment (JustAddWires) [inline source/BtI2cController.h]: Fixed.

File: source/CY8C9540ASupport.cpp (hunk 2 above)
Comment (JustAddWires) [inline source/CY8C9540ASupport.cpp]: Fixed.

File: source/PCA9685Support.cpp (hunk 7 above)
Comment (JustAddWires) [inline source/PCA9685Support.cpp]: Fixed.

File: source/PCAL9535ASupport.cpp (hunk 8 above)
Comment (JustAddWires) [inline source/PCAL9535ASupport.cpp]: Fixed.

**Pull request merged**